align/pdk/finfet/gen_param.py

```diff
@@ -41,23 +68,55 @@ def add_primitive(primitives, block_name, block_args):
                 if y == 1:
                     break
             pairs = limit_pairs((pairs))
+
```

Comment:
Should `limit_pairs` applied after legalization as legalization may eliminate a number of pairs anyway?